integration_test/testutil_test.go

```diff
@@ -183,12 +183,23 @@ func (it *ITest) Root() string {
 	return it.tempDir.Root()
 }
 
-// WithIndex initializes the index with the actual krew-index from github/kubernetes-sigs/krew-index.
-func (it *ITest) WithIndex() *ITest {
+// WithDefaultIndex initializes the index with the actual krew-index from github/kubernetes-sigs/krew-index.
+func (it *ITest) WithDefaultIndex() *ITest {
 	it.initializeIndex()
 	return it
 }
 
+// WithCustomIndexFromDefault initializes a new index by cloning the default index. WithDefaultIndex needs
+// to be called before this function.
```

Comment:
I think it's worth explaining _why_ we want this testutil in the comment. This is another good reason to move this logic into a function, because it gives us a good place where we can document this.